test/uk/gov/hmrc/fileupload/read/envelope/ServiceSpec.scala

```diff
@@ -68,4 +69,27 @@ class ServiceSpec extends UnitSpec with ScalaFutures {
       result shouldBe Xor.Right(file)
     }
   }
+
+  "regex for envelope size input" should {
+    "be true if enter is a valid size between 1kb to 25mb" in {
+      NotCreated.isValidSize("25mb", "25mb") shouldBe true
+      NotCreated.isValidSize("1kb", "25mb") shouldBe true
+    }
+    "be false if enter is an invalid size no between 1kb to 25mb" in {
+      NotCreated.isValidSize("26mb", "25mb") shouldBe false
+      NotCreated.isValidSize("0kb", "25mb") shouldBe false
+      NotCreated.isValidSize("kb", "25mb") shouldBe false
+      NotCreated.isValidSize("foo", "25mb") shouldBe false
+    }
+  }
+
+  "regex for change size from string to bytes Long" should {
+    "return -1 if input is invalid, otherwise return a long number" in {
```

Comment:
should throw an exception instead of returning -1 in case of invalid constraint input